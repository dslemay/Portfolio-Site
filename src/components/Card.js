import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import themeUtils from './themeUtils';

const CardWrapper = styled.div`
  max-width: ${props => props.maxWidth};
  margin-bottom: 1.5rem;
  border: 1px solid #ccc;
  padding: 0.5rem 0.25rem;
  box-shadow: 3px 3px 5px ${themeUtils.mediumAccent};
  ${props => props.style};
`;

const Card = props => (
  <CardWrapper style={props.style} maxWidth={props.maxWidth}>
    {props.children}
  </CardWrapper>
);

Card.propTypes = {
  maxWidth: PropTypes.string,
  style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  children: PropTypes.node.isRequired,
};

Card.defaultProps = {
  maxWidth: '',
  style: {},
};

export default Card;
