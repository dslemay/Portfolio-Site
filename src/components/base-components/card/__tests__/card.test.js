import React from 'react'
import { render } from 'react-testing-library'

import Card from '..'

describe('<Card />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Card>
        <div>Hello World</div>
      </Card>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
