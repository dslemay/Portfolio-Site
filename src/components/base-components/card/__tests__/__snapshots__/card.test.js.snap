// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<div
  class="css-17z98u"
>
  <div>
    Hello World
  </div>
</div>
`;
